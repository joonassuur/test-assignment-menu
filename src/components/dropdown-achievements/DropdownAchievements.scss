.dropdown-achievements {
  .title {
    font-size: 14px;
    &.main {
      font-size: 1em;
    }
  }
}
