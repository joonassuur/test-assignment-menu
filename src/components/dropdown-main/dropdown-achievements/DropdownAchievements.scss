.dropdown-achievements {
  .progress-container {
    align-items: center;
    display: flex;
    .progress {
      width: 80%;
    }
    .progress-count {
      width: 20%;
      text-align: right;
    }
  }
  .item:not(:first-child) {
    border-bottom: 1px solid #eeeef2;
    padding: 10px 0;
    margin: 0 16px;
  }
  .title,
  .subtitle {
    margin-bottom: 5px;
  }
  .title {
    font-size: 14px;
    &.main {
      font-size: 1em;
      .material-icons {
        cursor: pointer;
      }
    }
  }
  .title.main {
    display: flex;
    align-items: center;
    margin: 0;
    padding: 10px 16px;
    border-bottom: 1px solid #dcdce0;
    .material-icons {
      margin-right: 5px;
    }
  }
  .item.completed {
    .title,
    .progress-count {
      color: #37a447;
    }
    progress::-webkit-progress-value {
      background-color: #37a447;
    }
    progress::-webkit-progress-bar {
      background-color: #37a447;
    }
    progress::-moz-progress-bar {
      background-color: #37a447;
    }
  }
}
