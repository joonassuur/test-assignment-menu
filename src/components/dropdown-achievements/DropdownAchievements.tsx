import React from "react";
import { useDispatch } from "react-redux";
import { setActiveDD } from "../../redux/Index";
import "./DropdownAchievements.scss";

function DropdownAchievements() {
  const dispatch = useDispatch();

  const toggleAchievements = () => {
    dispatch(setActiveDD("general"));
  };

  return (
    <div className="dropdown-achievements">
      <div className="title main item" onClick={toggleAchievements} >Achievements</div>
      <div className="item">
        <div className="title">Choose crops</div>
        <div className="subtitle">
          Add crops to all of your fields in the current season
        </div>
        <progress
          className="progress small-bottom-margin"
          value="35"
          max="100"
        ></progress>
      </div>
      <div className="item">
        <div className="title">Add tasks to 3 fields</div>
        <div className="subtitle">
          Try adding tasks of your choise to 3 fields
        </div>
        <progress
          className="progress small-bottom-margin"
          value="33"
          max="100"
        ></progress>
      </div>
      <div className="item">
        <div className="title">Export the field diary report</div>
        <div className="subtitle">
          Get most important report in just few clicks :what:
        </div>
        <progress
          className="progress small-bottom-margin"
          value="0"
          max="100"
        ></progress>
      </div>
      <div className="item">
        <div className="title">Install mobile app</div>
        <div className="subtitle">
          See your farm data wherever you go, for you and for your workers
        </div>
        <progress
          className="progress small-bottom-margin"
          value="100"
          max="100"
        ></progress>
      </div>
      <div className="item">
        <div className="title">Add your employees</div>
      </div>
    </div>
  );
}

export default DropdownAchievements;
