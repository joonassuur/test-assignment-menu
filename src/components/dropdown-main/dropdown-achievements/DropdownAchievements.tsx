import React from "react";
import { useDispatch } from "react-redux";
import { setActiveDD } from "../../../redux/Index";
import "./DropdownAchievements.scss";

function DropdownAchievements() {
  const dispatch = useDispatch();

  const toggleAchievements = () => {
    dispatch(setActiveDD("general"));
  };

  return (
    <div className="dropdown-achievements">
      <div className="title main" onClick={toggleAchievements}>
        <span className="material-icons">arrow_back</span>
        <span>Achievements</span>
      </div>
      <div className="item">
        <div className="title">Choose crops</div>
        <div className="subtitle">
          Add crops to all of your fields in the current season
        </div>
        <div className="progress-container">
          <progress className="progress" value="35" max="100"></progress>
          <div className="progress-count subtitle">35%</div>
        </div>
      </div>
      <div className="item">
        <div className="title">Add tasks to 3 fields</div>
        <div className="subtitle">
          Try adding tasks of your choise to 3 fields
        </div>
        <div className="progress-container">
          <progress className="progress" value="33" max="100"></progress>
          <div className="progress-count subtitle">1/3</div>
        </div>
      </div>
      <div className="item">
        <div className="title">Export the field diary report</div>
        <div className="subtitle">
          Get most important report in just few clicks :what:
        </div>
        <div className="progress-container">
          <progress className="progress" value="0" max="1"></progress>
          <div className="progress-count subtitle">0/1</div>
        </div>
      </div>
      <div className="item completed">
        <div className="title">Install mobile app</div>
        <div className="subtitle">
          See your farm data wherever you go, for you and for your workers
        </div>
        <div className="progress-container">
          <progress className="progress" value="0" max="1"></progress>
          <div className="progress-count subtitle">
            <span className="material-icons-outlined">check</span>
          </div>
        </div>
      </div>
      <div className="item">
        <div className="title">Add your employees</div>
      </div>
    </div>
  );
}

export default DropdownAchievements;
