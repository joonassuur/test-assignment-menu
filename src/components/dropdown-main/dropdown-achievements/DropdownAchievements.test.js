import React from "react";
import TestRenderer from "react-test-renderer";

import DropdownAchievements from "./DropdownAchievements";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../../redux/Index");

describe("DropdownAchievements component", () => {
  test("Matches the snapshot", () => {
    const dropdownAchievements = TestRenderer.create(<DropdownAchievements />);
    expect(dropdownAchievements.toJSON()).toMatchSnapshot();
  });
});
