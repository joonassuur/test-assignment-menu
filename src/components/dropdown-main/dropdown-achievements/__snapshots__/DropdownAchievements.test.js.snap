// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownAchievements component Matches the snapshot 1`] = `
<div
  className="dropdown-achievements"
>
  <div
    className="title main"
    onClick={[Function]}
  >
    <span
      className="material-icons"
    >
      arrow_back
    </span>
    <span>
      Achievements
    </span>
  </div>
  <div
    className="item"
  >
    <div
      className="title"
    >
      Choose crops
    </div>
    <div
      className="subtitle"
    >
      Add crops to all of your fields in the current season
    </div>
    <div
      className="progress-container"
    >
      <progress
        className="progress"
        max="100"
        value="35"
      />
      <div
        className="progress-count subtitle"
      >
        35%
      </div>
    </div>
  </div>
  <div
    className="item"
  >
    <div
      className="title"
    >
      Add tasks to 3 fields
    </div>
    <div
      className="subtitle"
    >
      Try adding tasks of your choise to 3 fields
    </div>
    <div
      className="progress-container"
    >
      <progress
        className="progress"
        max="100"
        value="33"
      />
      <div
        className="progress-count subtitle"
      >
        1/3
      </div>
    </div>
  </div>
  <div
    className="item"
  >
    <div
      className="title"
    >
      Export the field diary report
    </div>
    <div
      className="subtitle"
    >
      Get most important report in just few clicks :what:
    </div>
    <div
      className="progress-container"
    >
      <progress
        className="progress"
        max="1"
        value="0"
      />
      <div
        className="progress-count subtitle"
      >
        0/1
      </div>
    </div>
  </div>
  <div
    className="item completed"
  >
    <div
      className="title"
    >
      Install mobile app
    </div>
    <div
      className="subtitle"
    >
      See your farm data wherever you go, for you and for your workers
    </div>
    <div
      className="progress-container"
    >
      <progress
        className="progress"
        max="1"
        value="0"
      />
      <div
        className="progress-count subtitle"
      >
        <span
          className="material-icons-outlined"
        >
          check
        </span>
      </div>
    </div>
  </div>
  <div
    className="item"
  >
    <div
      className="title"
    >
      Add your employees
    </div>
  </div>
</div>
`;
